dali/pipeline/operators/fused/crop_mirror_normalize.cc

```diff
@@ -56,92 +56,82 @@ normalization only.
 
 DALI_REGISTER_OPERATOR(CropMirrorNormalize, CropMirrorNormalize<CPUBackend>, CPU);
 
-namespace detail {
-
-template <typename OutputType, typename InputType>
-void RunHelper(Tensor<CPUBackend> &output,
-               const Tensor<CPUBackend> &input,
-               const std::vector<int64_t> &slice_anchor,
-               const std::vector<int64_t> &slice_shape,
-               bool horizontal_flip,
-               bool pad_output,
-               const std::vector<float> &mean,
-               const std::vector<float> &inv_std_dev) {
-  std::size_t number_of_dims = input.shape().size();
-  auto input_layout = input.GetLayout();
-  auto output_layout = output.GetLayout();
-  VALUE_SWITCH(number_of_dims, Dims, (3, 4), (
-    auto in_view = view<const InputType, Dims>(input);
-
-    kernels::SliceFlipNormalizePermutePadArgs<Dims> args(slice_shape);
-    for (std::size_t d = 0; d < Dims; d++) {
-      args.anchor[d] = slice_anchor[d];
-    }
-
-    if (pad_output) {
-      args.padded_shape[channels_dim(input_layout)] = 4;
-    }
-
-    if (horizontal_flip) {
-      args.flip[horizontal_dim_idx(input_layout)] = true;
-    }
-
-    // Check if permutation is needed
-    if (input_layout != output_layout) {
-      args.permuted_dims = permuted_dims<Dims>(input_layout, output_layout);
-    }
-
-    const bool should_normalize =
-         !std::all_of(mean.begin(), mean.end(), [](float x){ return x == 0.0f; })
-      || !std::all_of(inv_std_dev.begin(), inv_std_dev.end(), [](float x){ return x == 1.0f; });
-    if (should_normalize) {
-      args.mean = mean;
-      args.inv_stddev = inv_std_dev;
-      args.normalization_dim = channels_dim(input_layout);
-    }
-
-    kernels::SliceFlipNormalizePermuteCPU<OutputType, InputType, Dims> kernel;
-    kernels::KernelContext ctx;
-    kernels::KernelRequirements req = kernel.Setup(ctx, in_view, args);
-
-    output.set_type(TypeInfo::Create<OutputType>());
-    output.SetLayout(input.GetLayout());
-    output.Resize(req.output_shapes[0][0].shape.to_vector());
-
-    auto out_view = view<OutputType, Dims>(output);
-    kernel.Run(ctx, out_view, in_view, args);
-  ), // NOLINT
+template <>
+void CropMirrorNormalize<CPUBackend>::PrepareArgs(const HostWorkspace &ws,
```

Comment:
If I see correctly `ws` is not used so you can have one implementation for CPU and GPU as well.

Reply:
It was a leftover when I needed a TYPE_SWITCH here.